### ISSUE TITLE
Mouse Input Lag

### ISSUE THREAD

**Issue by Richard-L** [Return-To-The-Roots/s25client#1156]
Experiencing mild mouse input lag which is quite offputting.

What is the mouse polling rate in RttR? It should be 125hz minimum.

v191223 nightly

**Comment by Flamefire** [comment]
We are using SDL with push events so we have no real influence over that. However it is limited by the refresh rate/FPS as real updates are coupled with frames.

BTW: Out of interest: Why 125Hz when there are only 60FPS (or so)? You wouldn't notice every second update at all...

**Comment by Richard-L** [comment]
Thanks for getting back.

125hz mean 8ms input lag that are then "draped" onto the 60fps. It's usually fine, but 1000hz even cheap mice can do if the OS allows it. Back in the day with WinXP and even Win7 this was much easier to adjust. With a higher polling rate, the game benefits even with lower FPS.

These things make a big difference in competitive gaming, which is an aspect of RttR MP.

**Comment by Richard-L** [comment]
I had a dynamic framerate limiter set before. If disabled the responsiveness is better, albeit not good.

**Comment by Spikeone** [comment]
Well, using his mouse HZ Tester, I got 500Hz and it is working great

**Comment by Richard-L** [comment]
But that's only with _your_ mouse?

@Flamefire perhaps can you elaborate on how push based works? Is there no polling at some point checking whether a new push exists?

**Comment by Flamefire** [comment]
It basically is:

```
while True:
    while(any-event-found):
        update_state(event) (e.g. move cursor)
    draw()
```

Event queueing is handled by SDL which usually uses Windows messages (WM_MOUSEMOVE), so whatever the OS gives us is used, then the current state is drawn. We probably can't influence this. I can imagine there is some cut-off point, otherwise the inner loop would run forever if you constantly move the mouse.

Make sure you use SDL2 as your videodriver in RttR (see settings screen), which might have already improved this.

**Comment by Spikeone** [comment]
@Richard-L Did you try SDL2?

**Comment by Richard-L** [comment]
Yeah but it didn't change anything. Only the frame rate limiter had a significant effect, as mentioned above. You'd think that more frames rendered mean the moment it draws your mouse position is more precise, thus I had it disabled.

With the frame rate limited the responsiveness is much better.
